students/Jaidjen/lesson04/Mailroom2.py

```diff
@@ -1,77 +1,103 @@
-        print('Thank you', {new_mail}, 'for donating', {dnr_amt})
-    else:
-        dnr_amt = input("Please enter the donation amount: ")
-        Mail['Name']=new_mail
-        Mail['Donation']=dnr_amt
-        print('Dear',new_mail,',')
-        print('Thank you', new_mail,'for donating','{:.2f}'.format(int(dnr_amt)))
-        print('We appreciate your generosity.')
-        print('Sincerely, Donation Team')
+    while True:
+        response = input("Please enter the full name of the donor or select list? ")
+        if response == "list":
+            list_donors()
+            break
+        else:
+            amount = (int(input("Please enter the amount of donation: ")))
+            for donor in donors:
+                if donor["Name"] == response:
+                    donors.append((response, [amount]))
+                    break
+            else:
+                donors.append((response, [amount]))
+                print('Dear', response, ',')
+                print('Thank you', response, 'for donating', '{:.2f}'.format(int(amount)))
+                print('We appreciate your generosity.')
+                print()
+                print('Sincerely, Donation Team')
+                print()
+                break
 
+def list_donors():
+    for i in range(len(donors)):
+        print("{:<20s}".format(donors[i]["Name"]))
 
-def thankyou_mail():
-    for mail in Mail:
-        print('Dear',(Mail[mail]['Name']),',')
-        print('Thank you for your very kind donation of','{:.2f}'.format(sum(Mail[mail]['Donation'])))
+def thankyou_donors():
+    for donor in donors:
+        donations = donor["Donation"]
+        donated = sum(donations)
+        donor_name = donor["Name"]
+        print('Dear',(donor_name),',')
+        print('Thank you for your very kind donation of','{:.2f}'.format(donated))
         print('It will be put to very good use.')
         print('Sincerely, Donation Team')
         print(  )
 
 
+
 def exit_program():
     print("Bye!")
     sys.exit()  # exit the interactive script
 
 
 def main():
-    while True:
-        response = input(prompt)  # continuously collect user selection
-        # now redirect to feature functions based on the user selection
-        if response == "1":
-            add_mail()
-        elif response == "2":
-            view_mail(Mail)
-        elif response == "3":
-            thankyou_mail()
-        elif response == "4":
-            exit_program()
+    prompt = 0
+    while (prompt != 4):
+        print("1. Send a Thank You to a single donor")
+        print("2. Create a Report")
+        print("3. Send letters to all donors")
+        print("4. Quit")
+        prompt = int(input("Select a menu option: "))
+
+        if prompt == 1:
+            add_donors()
+        elif prompt == 2:
+            view_donors(donors)
+        elif prompt == 3:
+            thankyou_donors()
+        elif prompt == 4:
+            exit_program
```

Comment:
you should be using a switch dict in lesson 4 - do you need help how to apply that concept to the mailroom?